src/core/internal/fetch/fetchClient.ts

```diff
@@ -35,15 +40,15 @@ export class FetchClient implements HttpClient {
         config?: { headers: Record<string, string> },
         data?: unknown,
     ): Promise<T> {
-        let body;
+        let body = '{}';
```

Comment:
using proper `fetch` types uncovered a few minor TS issues, thus we see some minor edits below